src/NuGet.Core/NuGet.Protocol/Utility/MetadataReferenceCache.cs

```diff
@@ -97,12 +100,28 @@ public T GetObject<T>(T input)
                 _propertyCache.Add(typeKey, properties);
             }
 
+            if (!_stringMethodsCache.ContainsKey(typeof(MetadataReferenceCache)))
+            {
+                // Doing reflection everytime is expensive so cache it for string type which is all this MetadataReferenceCache about.
+                Type stringPropertyType = typeof(string);
+                MethodInfo method = _metadataReferenceCacheType.GetTypeInfo()
+                        .DeclaredMethods.FirstOrDefault(
+                            m =>
+                                m.Name == CachableTypesMap[stringPropertyType] &&
+                                m.GetParameters().Select(p => p.ParameterType).SequenceEqual(new Type[] { stringPropertyType }));
+                _stringMethodsCache.Add(_metadataReferenceCacheType, method);
```

Comment:
This doesn't look right to me. It's looking for methods defined on the `MetadataReferenceCache` class, rather than generic type `T`'s class.  Is that a mistake?

Reply:
I didn't add it it was there actually before my change, I just moved to top for multiple use.
![image](https://user-images.githubusercontent.com/8766776/85607861-5e395d00-b609-11ea-8c5a-3cd29a652efc.png)